### ISSUE TITLE
frida cant see all classes

### ISSUE THREAD

**Issue by rst630** [frida/frida#1134]
Fetched the list of classes by injecting:
```
Java.perform(function() {
    Java.enumerateLoadedClasses({
        onMatch: function(className) {
            console.log(className);
        }
    });
});
```

but in this list I can't find classes which I saw after dex2jar - sometimes in dex class names "a.a.d.e", "a.c.f" etc...

Some classes which cant see frida have real names like "com.appname.etc"

how I can inject method of this classes?

when I trying `Java.use("a.a.f.e.c");` it says that don't know this class

Spent 24h for solving this and no luck (( please help me

**Comment by BitTheByte** [comment]
same here. maybe it is something related to the new release

**Comment by oleavr** [comment]
Are you sure these classes are actually *loaded*? Classes are loaded the first time they're used, as far as I can recall.

**Comment by mh-** [comment]
Yes, I'm sure the classes are loaded, because they are used and executed, I can observe that from the behaviour of the app. 
But `frida-trace -U com.appname -I "*.a"` or `frida-trace -U com.appname -i "a"` doesn't find anything: "Started tracing 0 functions."

**Comment by Forterc** [comment]
@oleavr what's a good way to iterate over existing classes ( i.e not loaded )?
